python-avd/pyavd/_eos_designs/structured_config/network_services/utils_wan.py

```diff
@@ -474,45 +475,43 @@ def get_internet_exit_nat_acl_name(self: AvdStructuredConfigNetworkServicesProto
     def get_internet_exit_nat_pool_and_profile(
         self: AvdStructuredConfigNetworkServicesProtocol,
         internet_exit_policy_type: Literal["zscaler", "direct"],
-    ) -> tuple[dict | None, dict | None]:
+    ) -> tuple[EosCliConfigGen.IpNat.PoolsItem | None, EosCliConfigGen.IpNat.ProfilesItem | None]:
         if internet_exit_policy_type == "zscaler":
-            pool = {
-                "name": "PORT-ONLY-POOL",
-                "type": "port-only",
-                "ranges": [
-                    {
-                        "first_port": 1500,
-                        "last_port": 65535,
-                    },
-                ],
-            }
+            pool = EosCliConfigGen.IpNat.PoolsItem(
+                name="PORT-ONLY-POOL",
+                type="port-only",
+                ranges=EosCliConfigGen.IpNat.PoolsItem.Ranges([EosCliConfigGen.IpNat.PoolsItem.RangesItem(first_port=1500, last_port=65535)]),
+            )
 
-            profile = {
-                "name": self.get_internet_exit_nat_profile_name(internet_exit_policy_type),
-                "source": {
-                    "dynamic": [
-                        {
-                            "access_list": self.get_internet_exit_nat_acl_name(internet_exit_policy_type),
-                            "pool_name": "PORT-ONLY-POOL",
-                            "nat_type": "pool",
-                        },
-                    ],
-                },
-            }
+            profile = EosCliConfigGen.IpNat.ProfilesItem(
+                name=self.get_internet_exit_nat_profile_name(internet_exit_policy_type),
+                source=EosCliConfigGen.IpNat.ProfilesItem.Source(
+                    dynamic=EosCliConfigGen.IpNat.ProfilesItem.Source.Dynamic(
+                        [
+                            EosCliConfigGen.IpNat.ProfilesItem.Source.DynamicItem(
+                                access_list=self.get_internet_exit_nat_acl_name(internet_exit_policy_type),
+                                pool_name="PORT-ONLY-POOL",
+                                nat_type="pool",
+                            ),
+                        ]
+                    )
+                ),
+            )
```

Comment:
When you have deep nested models it can sometimes be nicer and more readable to first create the base class and then update it with the various shortcuts. Like:
```suggestion
            profile = EosCliConfigGen.IpNat.ProfilesItem(name=self.get_internet_exit_nat_profile_name(internet_exit_policy_type))
            profile.source.dynamic.append_new(
                access_list=self.get_internet_exit_nat_acl_name(internet_exit_policy_type),
                pool_name="PORT-ONLY-POOL",
                nat_type="pool",
            )
```